clients/drcachesim/common/options.cpp

```diff
@@ -749,6 +749,24 @@ droption_t<bool> op_record_replace_retaddr(
     "replacement, which has lower overhead, but runs the risk of breaking an "
     "application that examines or changes its own return addresses in the recorded "
     "functions.");
+droption_t<std::string> op_record_syscall(
+    DROPTION_SCOPE_CLIENT, "record_syscall", DROPTION_FLAG_ACCUMULATE,
+    OP_RECORD_FUNC_ITEM_SEP, "", "Record parameters for the specified syscall number(s).",
+    "Record the parameters and success of the specified system call number(s)."
+    " The option value should fit this format:"
+    " sycsall_number|parameter_number"
+    " E.g., -record_syscall \"2|2\" will record SYS_open's 2 parameters and whether"
+    " successful (1 for success or 0 for failure, in a function return value record)"
+    " for x86 Linux.  SYS_futex is recorded by default on Linux and this option's value"
+    " adds to futex rather than replacing it (setting futex to 0 parameters disables)."
```

Comment:
> setting futex to 0 parameters disables

I would expect setting the number of parameters to just reduce the num of parameters recorded but not disable it entirely. Since it's documented, it's probably okay. But do we want to support gathering only system call success and id vals (maybe the user doesn't want the syscall params for some reason?).

Reply:
It is a little weird but I liked the idea of having futex be on by default without trying to A) get the SYS_ value for all OS's inside options.cpp for a default value and B) the user having to re-specify futex to avoid disabling when trying to just add a syscall.  Alternatives could be a separate -record_futex parameter or having a parameter count of -1 disabling?? I would lean toward as-is with 0 disabling; it doesn't seem likely you would just want success.